pandas/core/arrays/masked.py

```diff
@@ -229,6 +231,30 @@ def to_numpy(
             data = self._data.astype(dtype, copy=copy)
         return data
 
+    def astype(self, dtype: Dtype, copy: bool = True) -> ArrayLike:
+        dtype = pandas_dtype(dtype)
+
+        if is_dtype_equal(dtype, self.dtype):
```

Comment:
here could in theory actually use `if self.dtype == dtype` since we know we have an extension dtype, and those should never raise when being compared (avoiding yet another set of checks, not sure if it would ever be significant, though).

Reply:
im fine with it either way.  this branch was broken off from one that was trying to push some astype boilerplate into a decorator so used the is_dtype_equal version